resources/migrations/001_update_migrations.yaml

```diff
@@ -8566,6 +8566,40 @@ databaseChangeLog:
             columnName: field_id
             remarks: This value is null if the field does not exist, or is created within a model
 
+  - changeSet:
+      id: v51.2024-07-29T16:00:00
+      author: metamben
+      comment: Add source card reference to report_card to support metrics based models and questions
+      preConditions:
+        - not:
+          - columnExists:
+              tableName: report_card
+              columnName: source_card_id
```

Comment:
As part of our query analysis and validation project, we recently added the `query_table` column which tracks which tables are read by a given query. In future we plan to extend this to also join to cards that are used as source models.

I don't know much about metrics, but do you think it makes sense to leverage this more generic table instead? One issue with analysis however is that we populate it asynchronously, so this reference could be stale.

Reply:
Here is the db schema we're imagining in future, however for how we've put off adding the `type` columns, and assume the data source is always a database table.

Reply:
For metric cards, we need the link to be set synchronously, for correctness.